### PR TITLE
fix: update broken links

### PR DESCRIPTION
**Pull request by MexicanAce**
# Description

Links are broken for the `era-compiler-solidity` repo, as it looks like that repo was refactored

### CODE REVIEW

File: content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md [hunk 1]
```diff
@@ -365,5 +365,5 @@ contract, its YUL code, assembly), the hash value may be different.
 
 Currently either the hash is computed as `keccak256`, or it is omitted completely.
 
-- [Definition of metadata hash.](https://github.com/matter-labs/era-compiler-solidity/blob/main/src/project/contract/metadata.rs)
-- [Front-end code that passes hash to the assembly.](https://github.com/matter-labs/era-compiler-solidity/blob/main/src/project/contract/mod.rs#L146)
+- [Definition of metadata hash.](https://github.com/matter-labs/era-compiler-solidity/blob/main/era-compiler-solidity/src/project/contract/metadata.rs)
```
Comment (hedgar2017) [inline content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md]: This is the definition of metadata hash *preimage*, not the hash itself.

File: content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md [hunk 2]
```diff
@@ -365,5 +365,5 @@ contract, its YUL code, assembly), the hash value may be different.
...
+- [Front-end code that passes hash to the assembly.](https://github.com/matter-labs/era-compiler-solidity/blob/main/era-compiler-solidity/src/project/contract/mod.rs#L106)
```
Comment (hedgar2017) [inline content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md]: Not to the *assembly*, but simply further down the compilation pipeline.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
Honestly, I don't think these docs and especially source code links are useful to anyone.

They shall be superseded by [zksolc CLI docs](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md) where we'll be able to keep them up-to-date.

**Comment by hedgar2017** [comment]
@MexicanAce let me describe it properly in zksolc docs, and I'll provide a link to be inserted here shortly.

**Comment by MexicanAce** [comment]
Copy. @hedgar2017 perhaps you wouldn't mind also reviewing this GitHub Discussion which lead me to these broken links: https://github.com/zkSync-Community-Hub/zksync-developers/discussions/784

Someone is trying to regenerate the hash at the end of `zksolc` bytecode from the metadata

**Comment by hedgar2017** [comment]
> Copy. @hedgar2017 perhaps you wouldn't mind also reviewing this GitHub Discussion which lead me to these broken links: https://github.com/zkSync-Community-Hub/zksync-developers/discussions/784
> 
> 
> 
> Someone is trying to regenerate the hash at the end of `zksolc` bytecode from the metadata

Sure, will do.

**Comment by hedgar2017** [comment]
@MexicanAce done!
Please use [this](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md#--metadata) and [this](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md#--metadata-hash) link instead.

File: content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md (hunk 1 above)
Comment (MexicanAce) [inline content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md]:
```suggestion
- [Definition of metadata hash.](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md#--metadata)
```

File: content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md [hunk 3]
```diff
@@ -365,5 +365,5 @@ contract, its YUL code, assembly), the hash value may be different.
 
 Currently either the hash is computed as `keccak256`, or it is omitted completely.
 
-- [Definition of metadata hash.](https://github.com/matter-labs/era-compiler-solidity/blob/main/src/project/contract/metadata.rs)
-- [Front-end code that passes hash to the assembly.](https://github.com/matter-labs/era-compiler-solidity/blob/main/src/project/contract/mod.rs#L146)
+- [Definition of metadata hash.](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md#--metadata)
+- [Front-end code that passes hash to the assembly.](https://github.com/matter-labs/era-compiler-solidity/blob/main/era-compiler-solidity/src/project/contract/mod.rs#L106)
```
Comment (MexicanAce) [inline content/10.zk-stack/10.components/70.compiler/20.specification/70.binary-layout.md]:
```suggestion
- [Front-end code that passes hash to the assembly.](https://github.com/matter-labs/era-compiler-solidity/blob/main/docs/src/02-command-line-interface.md#--metadata-hash)
```

**Comment by MexicanAce** [comment]
@hedgar2017 Links have been updated, please review the changes

**Review by uF4No — DISMISSED** [review]

**Comment by hedgar2017** [comment]
@MexicanAce made one more tweak! LGTM now.

**Review by hedgar2017 — APPROVED** [review]

**Pull request merged**